### PR TITLE
Add Snakemake

### PR DESCRIPTION
**Pull request by chrisburr**
Copies the `snakemake` and `snakemake-minimal` recipes from bioconda and merges them into a single multi-output recipe.

### CODE REVIEW

File: recipes/snakemake/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Copyright (c) 2016 Johannes Köster <johannes.koester@tu-dortmund.de>
```
Comment (chrisburr) [inline recipes/snakemake/LICENSE.md]: I've [made a PR upstream](https://bitbucket.org/snakemake/snakemake/pull-requests/375/add-licensemd-to-manifestin/diff) to include this in the PyPI tarball.

**Comment by chrisburr** [comment]
@johanneskoester I know quite a few people (including myself) who are using `snakemake` outside of bioinformatics. Would you mind if it was moved to `conda-forge`?

If yes, as you've been keeping the recipe updated in bioconda, would you like to be listed as a maintainer of the feedstock? I could also remove myself from the maintainer list if you'd prefer.

**Comment by kyleabeauchamp** [comment]
Previously, package switches between channels have been disruptive of people trying to maintain reproducible workflows.  I think we ought to focus on aligning the long-term strategy, maintenance, and accessibility of the two channels, rather than switching packages back and forth.  Just my two cents.

**Comment by chrisburr** [comment]
@conda-forge/bioconda-recipes any thoughts?

**Comment by kyleabeauchamp** [comment]
You can ask on the gitter if you want to poll random people's opinions: https://gitter.im/bioconda/Lobby

**Comment by bgruening** [comment]
@johanneskoester should decide that. I will ping him offline as well.

**Comment by johanneskoester** [comment]
Sorry, but i want snakemake to stay in bioconda exclusively. First, download counts, second I don't want people to geht confused, third reproducibility. There should be no problem to install snakemake from there even if people are not using it for bioinformatics.

**Comment by chrisburr** [comment]
I'm not going to push this but I'd like to make an attempt to persuade you and if you still disagree then that's fine and this PR can be closed.

I am opposed to recommending[1] people use `bioconda` as in my experience managing multiple channels confuses people especially given the order of channels needs to be correct. If Snakemake is the only reason for adding `bioconda` to the channel list I will probably suggest people use pip instead; though then the dependency management can get confusing.

In direct response to your concerns:

1. **Download counts**: These are currently split between the `bioconda` channel and PyPI so there already isn't one place for this number.
2. **User confusion**: As bioconda users are already using to using two custom channels I'm not sure how much confusion this would cause.
3. **Reproducibility**: I agree this very important.[2] Can this issue be avoided by:
    - copying all the binaries from `bioconda` to `conda-forge` so every version is still available
    - leave the existing binaries in the `bioconda` channel so people can re-create their old environments regardless of their channel configuration
    - `conda-forge` should already have a higher priority so new releases will be picked up from there

[1] Obviously this only applies to the fields I am a part of, it looks amazing for people who are doing bioinformatics research.
[2] Coincidentally I spent an hour yesterday figuring out why my Snakefile stopped working after an update. Though it seems to be much faster at building the DAG now so it was worth it!

**Comment by chrisburr** [comment]
Another option would be to have all binaries uploaded to both channels either from a conda-forge feedstock or the main bioconda repository? The downsides being that the download count is split further and it's slightly more complicated to set up and maintain.

**Comment by johanneskoester** [comment]
@chrisburr, you are right in the sense that there having to use another channel to install snakemake can be a burden for people outside of bioinformatics. To me though it feels like the impact is minor. 
For example, when defining an envioronment, you can simply prefix snakemake with `bioconda::`, no need to put the channel into to list of channels.
Instead, when choosing to put bioconda in the list of channels, conda can become a bit slower, however, that is something that has to be fixed in Conda in general (and we are working on this), not via moving packages between channels.

For the other arguments:
1. Yes there is a split between pypi and bioconda, but I don't want to make it even worse.
2 and 3. People will get confused by a sudden change. They will try to install Snakemake from bioconda and never see the newer versions. Although the bioconda manual clearly states the channel order, people are using all kinds of channel orders in reality.

So, sorry, but I think I would like to keep it as is.

**Comment by chrisburr** [comment]
Ah I didn't know channels could be prefixed like `bioconda::snakemake`. I think that makes it much less confusing and error prone to use bioconda for a single package. 👍

Thank you for taking the time to reply (and pointing out the `::` syntax), I'll close this now.

**Comment by johanneskoester** [comment]
The upcoming version of the docs now contains the rationale and the `bioconda::` hint. Thanks again for drawing my attention to this issue!

**Pull request closed without merging**